### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by pabloalarconm**
Missing ShEx files based on genetic diagnosis & Undiagnosed models

### CODE REVIEW

File: shex/sio-model/undiagnosedShape.shex [hunk 1]
```diff
@@ -0,0 +1,40 @@
+PREFIX : <http://purl.org/ejp-rd/cde/v020/shex/>
+PREFIX obo: <http://purl.obolibrary.org/obo/> 
+PREFIX sio: <http://semanticscience.org/resource/>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#> 
+
+:personShape IRI { 
+  a [ sio:SIO_000498 ] ;
+  sio:SIO_000228 @:role1Shape ;
```
Comment (sxzhang1201) [inline shex/sio-model/undiagnosedShape.shex:9]: Just wonder if it is possible to generalize them into one `roleShape,` and that `roleShape` allows `sio:is-realized-in` with both `healthcareActivityProcessShape` and `geneticTestingProcessShape`. How do you think @rajaram5 ?

File: shex/sio-model/undiagnosedShape.shex [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  sio:SIO_000228 @:role2Shape
+}
+
+:role1Shape IRI {
+  a [ obo:OBI_0000093 ] ;
+  sio:SIO_000356 @:process1Shape
+}
+
+:process1Shape IRI {
```
Comment (sxzhang1201) [inline shex/sio-model/undiagnosedShape.shex:18]: Would be great if it has a more semantic name, for example, `geneticTestingProcessShape`.

File: shex/sio-model/undiagnosedShape.shex [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  a [ obo:VariO_0138 ] ;
+  rdfs:label xsd:string
+}
+
+:role2Shape IRI {
+  a [ obo:OBI_0000093 ] ;
+  sio:SIO_000356 @:process2Shape
+}
+
+:process2Shape IRI {
```
Comment (sxzhang1201) [inline shex/sio-model/undiagnosedShape.shex:33]: Would be great if it has a more semantic name, for example, `healthcareActivityProcessShape`.

**Review by sxzhang1201 — CHANGES_REQUESTED** [review]
Both ShEx are syntactically correct and successfully validate the given example data. But the naming of shapes may need changes.

File: shex/sio-model/undiagnosedShape.shex (hunk 1 above)
Comment (sxzhang1201) [reply shex/sio-model/undiagnosedShape.shex:9]: Or we may need to distinguish `role1` and `role2` in a semantic way. Now both roles (https://github.com/ejp-rd-vp/CDE-semantic-model/blob/develop/images/sio-model/7_Undiagnosed.png) are a type of `obo: patient-role` and `sio:is-realized-in` processes, not much different from each other.

Comment (rajaram5) [reply shex/sio-model/undiagnosedShape.shex:9]: @sxzhang1201 sounds logical to be. maybe we can indicate that a person can have more than one role

File: shex/sio-model/undiagnosedShape.shex [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  a [ obo:NCIT_C16205 ] ;
+  sio:SIO_000229  @:phenotypeShape
+}
+
+:phenotypeShape IRI {
+  a [ obo:HP_0001249 ]
```
Comment (rajaram5) [inline shex/sio-model/undiagnosedShape.shex:39]: This HPO uri is used as an example in the examle ttl file. Can you replace it with regular expression?

File: shex/sio-model/undiagnosedShape.shex (hunk 2 above)
Comment (rajaram5) [reply shex/sio-model/undiagnosedShape.shex:18]: +1

File: shex/sio-model/undiagnosedShape.shex (hunk 3 above)
Comment (rajaram5) [reply shex/sio-model/undiagnosedShape.shex:33]: +1

**Review by rajaram5 — CHANGES_REQUESTED** [review]
Minor comments

File: shex/sio-model/undiagnosedShape.shex (hunk 1 above)
Comment (sxzhang1201) [reply shex/sio-model/undiagnosedShape.shex:9]: @rajaram5 I understand that a person can have more than one role. What I mean here is that we cannot tell any difference between `role1Shape` and `role2Shape`, but can tell between `healthcareActivityPatientRoleShape` and `geneticTestingPatientRoleShape`.

File: shex/sio-model/undiagnosedShape.shex (hunk 4 above)
Comment (pabloalarconm) [reply shex/sio-model/undiagnosedShape.shex:39]: Sure, the use of HPO ontology should be included at this Regular expression, or this shape could have any different type of OBO terms?

Comment (rajaram5) [reply shex/sio-model/undiagnosedShape.shex:39]: Regular expression of HPO is enough

File: shex/sio-model/undiagnosedShape.shex (hunk 1 above)
Comment (pabloalarconm) [reply shex/sio-model/undiagnosedShape.shex:9]: ```role1Shape``` corresponds to ``` healthcareActivityPatientRoleShape``` or  ```geneticTestingPatientRoleShape```?

Comment (pabloalarconm) [reply shex/sio-model/undiagnosedShape.shex:9]: ```healthcareActivityPatientRoleShape``` seems to be more involved with ```phenotypeShape``` and ```geneticTestingProcessShape``` with ```variantShape``` but the conversation seems to imply the other way around

**Review by rajaram5 — APPROVED** [review]
Looks good I will merge it

**Pull request merged**